test/FunctionalTests/Microsoft.OData.Core.Tests/ODataBinaryStreamReaderTests.cs

```diff
@@ -0,0 +1,71 @@
+﻿//---------------------------------------------------------------------
+// <copyright file="ODataBinaryStreamReaderTests.cs" company="Microsoft">
+//      Copyright (C) Microsoft Corporation. All rights reserved. See License.txt in the project root for license information.
+// </copyright>
+//---------------------------------------------------------------------
+
+using System;
+using System.IO;
+using System.Text;
+using System.Threading.Tasks;
+using Xunit;
+
+namespace Microsoft.OData.Tests
+{
+    /// <summary>
+    /// Unit tests for <see cref="ODataBinaryStreamReader"/>.
+    /// </summary>
+    public class ODataBinaryStreamReaderTests
+    {
+        private MemoryStream stream;
+        private const int maxLength = 14;
+
+        public ODataBinaryStreamReaderTests()
+        {
+            this.stream = new MemoryStream(Encoding.UTF8.GetBytes("The quick brown fox jumps over the lazy dog."));
+            this.stream.Position = 16;
+        }
+
+        private int ReadChars(char[] chars, int offset, int maxLength)
+        {
+            var buffer = new byte[maxLength];
+            int bytesRead = this.stream.Read(buffer, offset, maxLength);
+            buffer.CopyTo(chars, 0);
+
+            return bytesRead;
+        }
+
+        private async Task<int> ReadCharsAsync(char[] chars, int offset, int maxLength)
+        {
+            var buffer = new byte[maxLength];
+            int bytesRead = await this.stream.ReadAsync(buffer, offset, maxLength);
+            buffer.CopyTo(chars, 0);
+
+            return bytesRead;
+        }
+
+        //[Fact]
+        //public void ODataTextStreamReaderShouldRead()
+        //{
+        //    var reader = new ODataBinaryStreamReader(new Func<char[], int, int, int>(this.ReadChars));

```

Comment:
I know that these are commented out, but `reader` in both methods should be wrapped in a `using` block